File: week3/Choi/Week3/Mocha.cs
using System;

namespace Week3
{
    public class Mocha : CondimentDecorator
    {
        public Beverage m_Beverage;

        public Mocha(Beverage ver)
        {
         
            m_Beverage = ver;
        }

        public override double Cost()
        {
            return 0.2f + m_Beverage.Cost();
        }

        public override string GetDescription()
        {
            return m_Beverage.GetDescription() + " 모카";
        }
    }
}

File: week3/minj0i/Mocha.java
package week3.minj0i;

public class Mocha extends CondimentDecorator {
	Beverage beverage;
	
	public Mocha(Beverage beverage) {
		this.beverage = beverage;
	}
	
	public String getDescription() {
		return beverage.getDescription() + ", 모카";
	}
	
	public double cost() {
		return .20 + beverage.cost();
	}
}
